robot/impl/local_robot.go

```diff
@@ -937,6 +917,28 @@ func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config)
 		return
 	}
 
+	// If something was added, go through components and services in diff.Added,
+	// call Validate on all modularized resources, and store those resources'
+	// implicit dependencies.
+	if diff.Added != nil {
```

Comment:
why only on added?

Reply:
Good question; I think we should also look through `diff.Modified`. Previously, we were looking through the entire `newConfig` which was getting implicit dependencies for all components and services even if they hadn't been modified. Now, we recalculate implicit dependencies only if a component or service has been added or modified.